File: javascript/giorno3.js
/*
Trascrizione (best-effort) - somma lista

SOMMA = 0
FOR elem IN LISTA
  SOMMA = SOMMA + elem
PRINT(SOMMA)
*/

function sumList(nums) {
  let total = 0;
  for (const n of nums) total += n;
  return total;
}

if (require && require.main === module) {
  console.log(sumList([1, 2, 3]));
}

File: python/giorno3.py
"""
Trascrizione (best-effort) dall'immagine (esercizio somma/lista):

GIORNO 3 (SOMMA) LISTA 1,2,3
SOMMA = 0
FOR i IN LISTA
    SOMMA = SOMMA + i
PRINT(SOMMA)

Traduzione: funzione sum_list che calcola la somma degli elementi.
"""

from typing import List


def sum_list(nums: List[float]) -> float:
    total = 0
    for x in nums:
        total += x
    return total


if __name__ == "__main__":
    print(sum_list([1, 2, 3]))
